src/ray/gcs/redis_context.cc

```diff
@@ -384,6 +388,63 @@ Status RedisContext::PingPort(const std::string &address, int port) {
       address, port, redisConnect, static_cast<redisContext **>(nullptr));
 }
 
+void ValidateRedisDB(RedisContext &context) {
+  auto reply = context.RunArgvSync(std::vector<std::string>{"INFO", "CLUSTER"});
+  // cluster_state:ok
+  // cluster_slots_assigned:16384
+  // cluster_slots_ok:16384
+  // cluster_slots_pfail:0
+  // cluster_size:1
+  RAY_CHECK(reply && !reply->IsNil()) << "Failed to get Redis cluster info";
+  auto cluster_info = reply->ReadAsString();
+
+  std::vector<std::string> parts = absl::StrSplit(cluster_info, "\r\n");
+  bool cluster_mode = false;
+  int cluster_size = 0;
+
+  // Check the cluster status first
+  for (auto &part : parts) {
+    if (part.empty() || part[0] == '#') {
+      // it's a comment
+      continue;
+    }
+    std::vector<std::string> kv = absl::StrSplit(part, ":");
+    RAY_CHECK(kv.size() == 2);
+    if (kv[0] == "cluster_state") {
+      if (kv[1] == "ok") {
+        cluster_mode = true;
+      } else if (kv[1] == "fail") {
+        RAY_LOG(FATAL)
+            << "The Redis cluster is not healthy. cluster_state shows failed status: "
+            << cluster_info;
+      }
+    }
+    if (kv[0] == "cluster_size") {
+      cluster_size = std::stoi(kv[1]);
+    }
+  }
+
+  if (cluster_mode) {
+    RAY_CHECK(cluster_size == 1)
+        << "Ray currently doesn't support sharded Redis cluster. "
```

Comment:
```suggestion
        << "The Redis cluster size is " << cluster_size << ". Ray currently doesn't support sharded Redis cluster. "
```

Reply:
Maybe I am ignorant in Redis, but "not supporting shared cluster, and provide cluster with 1 shard" sounds contradicting. Does this mean we can have only 1 shard cluster? (master + 1 shard)

Reply:
Oh, 1 shard and for each shard you still can have multiple replicas.